common/voxel/chunkmanager.cpp

```diff
@@ -236,18 +362,81 @@ Vector3* ChunkManager::translateIndexToPosition(
             2); // Calculating the Z position of the Chunk in the World
   return new Vector3(ara[0], ara[1], ara[2]);
 }
+*/
+
+std::string ChunkManager::generateKey(int sx, int sy, int sz)//translates a chunk position value, to a standardized hash map key
+{
+	std::stringstream ssKey;				//creating a stringstream object to translate the integers to a string
+	ssKey << sx << "," << sy << "," << sz;	//standard format for the hash map keys is 'x,y,z'
+	return ssKey.str();						//returning the formatted string
+}
+
+Vector3 ChunkManager::keyToVector(std::string key)//translates a hash table key to a Vector3 for reduce operations
+{
+	std::vector<int> result;
+	std::stringstream data(key);
+	std::string line;
+	while(std::getline(data,line,','))
+	{
+		result.push_back(std::stoi(line));
+	}
+	float x_pos = result[0];
+	float y_pos = result[1];
+	float z_pos = result[2];
+	
+	Vector3 chunk(x_pos, y_pos, z_pos);
+	return chunk;
+}
+
```

Comment:
Clients don't send a string to the chunk manager they send a vector and we convert it to a string.  The client does not need to know how we track everything.  That is what 
 std::string generateKey(int x, int y, int z); 
 Vector3 keyToVector(std::string key); 
These handle the conversion.  The use of the string is a consequence of using a concurrent_hash_map.